sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/filtering/DerivedMetricProjections.java

```diff
@@ -0,0 +1,113 @@
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.swagger.models.DerivedMetricInfo;
+import com.google.common.util.concurrent.AtomicDouble;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class DerivedMetricProjections {
+
+    public static final String COUNT = "Count()";
+    private final Map<String, DerivedMetricAggregation> derivedMetricValues = new HashMap<>();
+
+    public DerivedMetricProjections(Map<String, AggregationType> projectionInfo) {
+        for (Map.Entry<String, AggregationType> entry : projectionInfo.entrySet()) {
+            AggregationType aggregationType = entry.getValue();
+            DerivedMetricAggregation value;
+            if (aggregationType.equals(AggregationType.MIN)) {
+                value = new DerivedMetricAggregation(Long.MAX_VALUE, aggregationType);
+            } else if (aggregationType.equals(AggregationType.MAX)) {
+                value = new DerivedMetricAggregation(Long.MIN_VALUE, aggregationType);
+            } else if (aggregationType.equals(AggregationType.SUM) || aggregationType.equals(AggregationType.AVG)) {
+                value = new DerivedMetricAggregation(0, aggregationType);
+            } else {
+                value = null; // we should never hit this case - that means the UI gave us an invalid aggregation type
+            }
+            derivedMetricValues.put(entry.getKey(), value);
+        }
+    }
+
+    // This is intended to be called once for every post request
+    public Map<String, Double> fetchFinalDerivedMetricValues() {
+        Map<String, Double> result = new HashMap<>();
+        for (Map.Entry<String, DerivedMetricAggregation> entry : derivedMetricValues.entrySet()) {
+            String id = entry.getKey();
+            DerivedMetricAggregation dma = entry.getValue();
+            double intermediateValue = dma.aggregation.get();
+            long count = dma.count.get();
+            if (count == 0) {
+                result.put(id, 0.0);
+            } else {
+                if (dma.aggregationType.equals(AggregationType.AVG)) {
+                    result.put(id, intermediateValue / count);
+                } else {
+                    result.put(id, intermediateValue);
+                }
+            }
+        }
+        return result;
+    }
+
+    // Once a telemetry item passes a metric chart filter, we use that telemetry item to increment
+    // a derived metric
+    public void calculateProjection(DerivedMetricInfo derivedMetricInfo, TelemetryColumns columns) {
+        double incrementBy = Double.NaN;
+        if (COUNT.equals(derivedMetricInfo.getProjection())) {
+            incrementBy = 1.0;
+        } else if (KnownRequestColumns.DURATION.equals(derivedMetricInfo.getProjection())) {
+            if (columns instanceof RequestDataColumns || columns instanceof DependencyDataColumns) {
```

Comment:
Could it be merged into the `else if`?
```suggestion
        } else if (KnownRequestColumns.DURATION.equals(derivedMetricInfo.getProjection())) {
            if (columns instanceof RequestDataColumns || columns instanceof DependencyDataColumns) {
```

Reply:
I am thinking of removing the inner if entirely, as the UI will guarantee that Duration projections are only possible for request / dependency type.